src/main/java/bio/terra/service/filedata/flight/ingest/FileIngestWorkerFlight.java

```diff
@@ -0,0 +1,73 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.app.configuration.ApplicationConfiguration;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.FileService;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.filedata.google.firestore.FireStoreUtils;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.resourcemanagement.DataLocationService;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.RetryRuleRandomBackoff;
+import bio.terra.stairway.UserRequestInfo;
+import org.springframework.context.ApplicationContext;
+
+import java.util.UUID;
+
+/*
+ * The flight is launched from the IngestDriverStep within one of the bulk load flights.
+ * It performs a single file ingest into a dataset.
+ * Input parameters expected:
+ * - DATASET_ID - dataset into which we load the file
+ * - REQUEST - a FileLoadModel describing the file to load
+*/
+
+public class FileIngestWorkerFlight extends Flight {
+
+    public FileIngestWorkerFlight(FlightMap inputParameters,
+                                  Object applicationContext,
+                                  UserRequestInfo userRequestInfo) {
+        super(inputParameters, applicationContext, userRequestInfo);
+
+        ApplicationContext appContext = (ApplicationContext) applicationContext;
+        FireStoreDao fileDao = (FireStoreDao)appContext.getBean("fireStoreDao");
+        FireStoreUtils fireStoreUtils = (FireStoreUtils)appContext.getBean("fireStoreUtils");
+        FileService fileService = (FileService)appContext.getBean("fileService");
+        GcsPdao gcsPdao = (GcsPdao)appContext.getBean("gcsPdao");
+        DatasetService datasetService = (DatasetService)appContext.getBean("datasetService");
+        DataLocationService locationService = (DataLocationService)appContext.getBean("dataLocationService");
+        ApplicationConfiguration appConfig =
+            (ApplicationConfiguration)appContext.getBean("applicationConfiguration");
+
+        UUID datasetId = UUID.fromString(inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), String.class));
```

Comment:
I started using String instead of UUID to eliminate converting back and forth all of the time. The result is inconsistent code and represents tech debt. FileIngestWorkerFlight is re-using existing code. FileArrayIngestFlight I just wrote and used Strings.

There is another ticket, DR-441, that the use of UUIDs. Subsequently, I learned about "short UUIDs" which take the binary UUID and encode it as a base-64ish string (22 chars) instead of 36 chars of a hex string with dashes UUID. My opinion is that we should remove all of our UUID use and make a class that extends String and generates short UUIDs. It would lower our row-id footprint in BigQuery by 38%...